### DISCUSSION TITLE
Prognose bei OST-West-Ausrichtung

### DISCUSSION THREAD

**Discussion by harrykausl** [wiggal/GEN24_Ladesteuerung#55] [General]
Ich habe eine Anlage (13,32 kW) mit Ost-West-Ausrichtung, d.h. 18 Module Ost und 18 West. Bei Solarcast und Forecast Solar sind dafür 2 Anlagen einzugeben, Solarprognose geht nur bis 10 kW. Wie schaffe ich es, die beiden zusammen prognostiziert zu bekommen. In den Skripten sieht es so aus, als ob es nur für eine Anlage funktioniert. Danke im Voraus.

**Comment by wiggal** [comment]
Solarcast:
hier kannst dur die beiden Ausrichtungen in der config.ini unter forecast.solar und forecast.solar2 konfigurieren

Solarprognose:
hier könntest du die beiden Ausrichtungen mit je 5kWp konfigurieren und in der config.ini mit KW_Faktor = 1.332 multiplizieren

Forecast Solar:
hier weis ich nicht, wie zwei Ausrichtungen zu konfigurieren sind, aber irgend jemand hat das schon mal geschrieben, dass dies gehen soll. 
siehe: https://solcast.com/free-rooftop-solar-forecasting

**Comment by harrykausl** [comment]
Danke!

**Reply by harrykausl** [reply]
Sorry, hat ein bisschen gedauert, musste erst noch meinen 2. pi freischaufeln. Ich habe bei solcast 2 Anlagen definiert Jetzt ist mir aber nicht klar, wie ich die in die config bekommen. 2 Anlagen sind da nur für forecast solar vorgesehen.

**Reply by wiggal** [reply]
Hi, ich hab keine Anlage mit zwei Ausrichtungen.
Bei Solcast kannst du nur eine API-Abfrage konfigurieren, wenn die für zwei Ausrichtungen bei solcast.com.au nicht zu konfigurieren ist, dann gehts nicht.

**Reply by harrykausl** [reply]
Schade und Danke!

**Reply by harrykausl** [reply]
Ich habe doch noch eine Frage. Ich habe in meiner Domoticz-Hausautomation die solcast-Vorhersage für Ost-/West mal rein programmiert (Ertrag und maximale Leistung heute und morgen). Das sind 4 api-Aufrufe (Ost, West, forecast, actual). Könnte / dürfte ich Deine symo-skripte abspecken, damit ich sie aufrufen kann. Ich bräuchte ja nur Steuerung ein und aus, sowie Lesen / Setzen der maximalen Ladeleistung, das zeitliche mache ich aus Domoticz. Hintergrund ist, dass ich meine Anlage erweitert habe (13kW) und der WR (Fronius Gen24 10.0) nur 10 kW wandeln kann. Im Sommer würde ich dann morgens die Ladung der Batterie begrenzen und zur Hochzeit die Batterie mit z.B. 3 KW laden, damit der WR nicht abregelt.  
Es müsste etwa so sein:

symno...py steuerung ein
symno...Py steuerung aus
symno..py read max_batterieload
symno..py read max_batterieload

**Reply by wiggal** [reply]
Natürlich kannst du mit den Skripten machen was du willst / brauchst, ist ja Opensource.
Viel Spaß, falls du Fragen hast, jeder Zeit.

**Reply by wiggal** [reply]
Übrigends kann der GEN24 maximal nur 11.4KW verarbeiten, wenn mindestens 1KW in die Batterie geht (10,4 kann er wandeln und zusätzlich 1KW DC in den Akku)

**Reply by harrykausl** [reply]
Das kommt aber bei Ost-/West Ausrichtung hoffentlich nicht so oft vor, dann lass ich den WR lieber abregeln anstatt einen 2. aufzustellen mit zusätzlicher Verkabelung.
Noch eine Frage. Wenn ich das weatherData json für solcast so erzeuge, wie Du es brauchst, könnte ich es doch Deinem Programm unterjubeln?

**Reply by wiggal** [reply]
Ja, wenn du die weatherData.json extern in dem erwartetem Format erstellst, arbeitet meine SymoGen24Controller2.py damit.
Die weatherData.json wird ja bei mir auch von extrigen Skripten erzeugt, wo die Datei herkommt, ist dem SymoGen24Controller2.py egal.

**Reply by harrykausl** [reply]
Hallo Wiggal,

danke für Deine Unterstützung, ein tolles Projekt.

Eigentlich würde es mir ja reichen, wenn ich Batterieregeln per api aktivieren / deaktivieren könnte. Das ist aber leider für die normalen Batterieregeln im WR nicht möglich, das geht nur auf der WEB-Seite. Was m.E. nicht schlecht wäre, wenn Dein Programm 2 Programmteile hätte, einen Berechnungsteil, dieser schreibt am Ende eine json oder xml-Auftragsdatei. Und einen Ausführungsteil, dem man den Namen der json oder xml mitgibt und der dann alles so setzt, wie es in der Datei steht. Man hätte dann den Vorteil, dass man unterschiedliche Rulesets definieren könnte unabhängig von der Vorhersage. Ein Ruleset wäre z.B. auch das Ganze zu deaktiveren.  Dann könnte man über den Aufruf des 2. Programmteiles mit Angabe der Eingabedatei ein vorgefertigtes Set aktivieren.
Ich muss mir mal anschauen, wie ich den 2. Teil aus Deinen Skripten herausziehen könnte.

PS: Wo sehe ich im Webinterface des WR die Daten, die gesetzt wurden. Unter Komponenten / Batterie?

**Reply by wiggal** [reply]
Hi,
gleich mal zum letzten Punkt, Modbuseinstellungen sieht man im Webinterface des WR  leider nicht, ist oftmals etwas verwirrend, wenn man nicht weiß welche Einstellungen nun gültig sind.

Dann zu deinem Wunsch mit der Steuerung:

Es gibt zwei Steuerdateien, die ich über das Webinterface meines Programmes ( LadeSteuerung und EntladeSteuerung) erzeuge, und das dann vom Script gelesen und verarbeitet wird. 
Die Dateien heißen  Watt_Reservierung.json und Akku_EntLadeSteuerFile.json, mit den Felder am Ende der Dateien  "ManuelleSteuerung" bzw. "ManuelleEntladesteuerung" werden feste Lade bzw. Entladewerte gesetzt.
Die könnte man auch mit externen Programmen erzeugen.
Weiß nicht, ob dir das weiterhilft

**Reply by harrykausl** [reply]
Danke, probiere ich aus,

**Reply by harrykausl** [reply]
Nachdem ich mir den Code durchgesehen habe doch noch eine Frage. Das herunter regeln der Batterieladung auf einen in WattReservierung angegebenen Wert  (z.B. wegen 70%-Einspeiseregelung ) geschieht durch setzen der maximalen prozentualen Ladung der Batterie und nicht durch setzen des Ladestroms wie in den WR-Batterieregeln? Wenn der Strom nicht begrenzt wird, bedeutet das dann, dass mit voller PV geladen wird, bis die Batterie die maximal eingestellten Prozent erreicht hat und es geht erst wieder weiter, wenn die Prozente erhöht werden? Danke im Voraus.

**Reply by wiggal** [reply]
Hi,
die Angabe "ManuelleSteuerung" in der WattReservierung.json ist der Anteil bezogen auf die Variable MaxLadung (in Watt) aus de config.ini.

z.B.:
config.ini:
MaxLadung = 3000 => die Ladedung über mein Programm beträgt höchstens 3000 Watt
WattReservierung.json:
"ManuelleSteuerung": 0.5 => die ladung erfolgt mit 3000W * 0.5 = 1500W

Ich hoffe das ist verständlich.

**Reply by harrykausl** [reply]
Kapiert, danke!

**Comment by zellneralex** [comment]
Ich schließe mich hier mal an da es mich auch betrifft. Zu Forecast Solar:
Um 2 Ausrichtungen abfragen zu können müssen 2 Sites angelegt werden die nicht weiter als 1km auseinander liegen dürfen.
![IMG_4562](https://github.com/wiggal/GEN24_Ladesteuerung/assets/37265689/b200caf5-459f-42cd-88b4-ee44529b0edb)
Am Limit von 5 Abfragen am Tag (da 2 API Anfragen gebraucht werden) ändert sich dadurch nichts da das Limit bei 10 API Abfragen pro Site und  Tag liegt.

Ich würde es in den nächsten Tagen mal aufbauen wie bei Solarcast. Und einfach einen 2ten config Block nur mit resource_id und KW_Faktor machen und das dann testen.

Ich würde daraus dann nen PR machen falls @wiggal sowas zulässt.

**Reply by wiggal** [reply]
Hi, hab natürlich nichts dagegen, dass du einen PR aufmachst, muss mich aber erst noch mit den Berechtigungen in Github befassen.

**Reply by zellneralex** [reply]
Das ist gar kein Problem, ich mache einen fork und wenn es bei mir läuft erzeuge ich einen pull request der dann wie deine in der Liste stehen. Und du als Maintainer must den dann mergen oder nicht.

**Reply by wiggal** [reply]
ok vielen Dank für die Info, hatte ich noch nie, war bisher immer alleine ;-)

**Comment by zellneralex** [comment]
Ok, ich habe gute und schlechte Nachricht:
Erst mal die guten: Ich kann nun beide Ausrichtungen monitoren, und nun kommt das aber, ich hab deren Aussage falsch interpretiert (Erst beim nächsten link wird klar das es maximal 10 API calls am Tag sind, egal ob 1 oder 2 Sites) -> Mit 2 Ausrichtungen kann man maximal 2 mal am Tag abfragen (4 API calls pro Abfrage)

Ich hab bisher [pv.strings].anzahl auch für SolarCast benutzt werde das aber jetzt noch mal umbauen und einen 2ten unabhängig Parameter anlegen so das der Benutzer entscheiden kann ob er eine genauere Abfrage mit 2 Ausrichtungen machen will oder wie auf der Webseite ebenfalls vorgeschlagen mit einem Feld und einem Kompromiss der Ausrichtungen.

Ich mach dann morgen mal nen Push der config so das du da mal drüber schauen kannst ob das für dich ok ist.

**Reply by wiggal** [reply]
Hi,
es wäre auch denkbar nur "forecasts" aufzurufen, da "estimated_actuals" ja nur die Vergangenheit abbildet.
"estimated_actuals" hat keinerlei Einfuss auf die Berechnungen, es würden halt nur in der Tabelle der Ladesteuerung die Prognosewerte der Vergangenheit fehlen.
Oder man fragt die Prognose ab und nimmt die Vergangenheit aus der "weatherData.json", wird halt komplizierter.

**Reply by zellneralex** [reply]
Muss ich mir mal anschauen was das für das logging bedeutet. Das json sieht halt dann so aus 
<img width="586" alt="image" src="https://github.com/wiggal/GEN24_Ladesteuerung/assets/37265689/fe5d62b4-b1c5-483e-a19f-be931babe652">
Das war die Abfrage um 11 als estimated_actuals mit Error zurück kam.

**Comment by wiggal** [comment]
ja ok, das reicht nicht, für die aktuelle Stunde brauchen wir Werte. Dann bliebe nur die historischen Werte aus dem weather.json File zu holen

Ich probiere es morgen mal nur mit derPrognose

**Reply by zellneralex** [reply]
Ok, bin gerade weg vom PC dann Check ich morgen mal ein was ich hab und schick nen link auf das repo

**Reply by zellneralex** [reply]
Guten Morgen, 

https://github.com/zellneralex/GEN24_Ladesteuerung/tree/main ist mein aktueller Stand. Schreib mir einfach ob du nur mit "forecasts" klar kommst. Sobald du das entweder in main oder einem eigenen Branch hast merge ich nochmal und lass es bei mir in der test Datei laufen.

Ich habe auch deine cronjob Empfehlung angepasst:

33 5,7,9,11,13,15,17 * * * /DIR/start_PythonScript.sh WeatherDataProvider2.py
8 5,7,9,11,13,15,17 * * * /DIR/start_PythonScript.sh Solarprognose_WeatherData.py
15 6,8,11,13,15 * * * /DIR/start_PythonScript.sh Solcast_WeatherData.py
und alle 3 dataAgeMaxInMinutes auf 100 gestellt. so sollte Solarprognose immer ein Wert liefern und falls das nicht so ist einer der anderen 2 eben dann die Daten schreiben. Das ist nicht drin da ich das erst mal eine Weile laufen lassen will.

**Reply by wiggal** [reply]
Servus,
ich bin in Git nicht so fit,
wie muss ich nun weiter machen, dass ich deine Änderungen in einen Branch bzw. einen PR bekomme?

**Reply by zellneralex** [reply]
Im Moment musst du noch gar nichts machen. Ich hab dir erst mal den link geschickt das du sehen kannst was ich getan habe.

Ich kann dir aber schnell einen PR erzeugen den du mergen kannst.

**Reply by wiggal** [reply]
Ja, mach mal eine PR, dann können wir dort weiter diskutieren, oder?

**Reply by zellneralex** [reply]
PR ist fertig https://github.com/wiggal/GEN24_Ladesteuerung/pull/57 Den kannst du jetzt ganz normal merrgen

**Comment by wiggal** [comment]
> Ich habe auch deine cronjob Empfehlung angepasst:
> 
> 33 5,7,9,11,13,15,17 * * * /DIR/start_PythonScript.sh WeatherDataProvider2.py 
> 8 5,7,9,11,13,15,17 * * * /DIR/start_PythonScript.sh Solarprognose_WeatherData.py 
> 15 6,8,11,13,15 * * * /DIR/start_PythonScript.sh Solcast_WeatherData.py 
> 
> und alle 3 dataAgeMaxInMinutes auf 100 gestellt.
> so sollte Solarprognose immer ein Wert liefern und falls das nicht so ist einer der anderen 2 eben dann die Daten schreiben. 
> Das ist nicht drin da ich das erst mal eine Weile laufen lassen will.

Da würde ich beachten, dass durch die 100 Minutengrenze evtl, die Prognose von verschiedenen APIs kommt.
Da die Prognosen teilweise stark abweichen, bekommst du  bei der Prognoseberechnung evtl. ein ständiges auf und ab.
Achte mal bei den Tests darauf.

**Reply by zellneralex** [reply]
Mit der Reihenfolge sollte immer Solarprognose “gewinnen” Nur wenn die offline wäre sollte einer der anderen beiden Daten schreiben. Ich werde aber darauf achten.

**Reply by wiggal** [reply]
Solarprognose_WeatherData holt die Daten um 5:08, wenn dann Solcast_WeatherData um 6:15 holen will sind noch keine 100 Minuten um, usw.
Deshalb habe ich die Backupseiten auf 1000 Minuten gestetzt, bzw. bei mir laufen die Backupseiten inzwischen nur einmal am Tag kurz nach Solcast_WeatherData.

**Reply by wiggal** [reply]
@harrykausl,
den PR von @zellneralex habe ich soeben eingearbeitet. 
Nun können auch bei Solcast zwei Ausrichtungen abgefragt werden.

Danke nochmal an @zellneralex
